src/main/java/io/debezium/connector/yugabytedb/SourceInfo.java

```diff
@@ -95,11 +95,19 @@ public final class SourceInfo extends BaseSourceInfo {
     private String tableName;
     private String tabletId;
 
+    // todo Vaibhav: maybe add another constructor in order to initialize the SourceContext properly
     protected SourceInfo(YugabyteDBConnectorConfig connectorConfig) {
         super(connectorConfig);
         this.dbName = connectorConfig.databaseName();
     }
 
+    protected SourceInfo(YugabyteDBConnectorConfig connectorConfig, OpId lastCommitLsn) {
```

Comment:
what will be the lsn if the original constructor is used?

Reply:
It will be null, but that constructor is only used at one place while starting the connector for the first time, after which the connector calls the function `SourceInfo#update()` to change all those values.

Also yes, I believe you're trying to indicate a potential `NullPointerException` if this is not coded properly, but this is again handled in the class `YugabyteDBOffsetContext`, see the below code snippet:
```java
OpId lsn() {
    return sourceInfo.lsn() == null ? new OpId(0, 0, null, 0, 0)
        : sourceInfo.lsn();
}
```

So in case the original constructor is used, we will try to simply ask the server to send the changes from the beginning, but the server will only send from the checkpoint which has not been read, and then when we get those new changes, we can simply update the lsn for the tablet.